public/index.html

```diff
@@ -1,44 +1,36 @@
-<!doctype html>
+<!DOCTYPE html>
 <html lang="en">
 
-<head>
-  <meta charset="utf-8">
-  <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
-  <!--
-      Notice the use of %PUBLIC_URL% in the tag above.
-      It will be replaced with the URL of the `public` folder during the build.
-      Only files inside the `public` folder can be referenced from the HTML.
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
+    <title>OpBeans</title>
+    <script type="text/javascript" src="/rum-config.js"></script>
+  </head>
 
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
-  <title>OpBeans</title>
-  <script type="text/javascript" src="/rum-config.js"></script>
-</head>
+  <body onload="init()">
+    <div id="root"></div>
+    <!--
+        This HTML file is a template.
+        If you open it directly in the browser, you will see an empty page.
 
-<body>
-  <div id="root"></div>
-  <!--
-      This HTML file is a template.
-      If you open it directly in the browser, you will see an empty page.
-
-      You can add webfonts, meta tags, or analytics to this file.
-      The build step will place the bundled scripts into the <body> tag.
-
-      To begin the development, run `npm start`.
-      To create a production bundle, use `npm run build`.
-    -->
-</body>
-<script>
-  var pageName = window.location.pathname.split('/')[1]
-  window.elasticApm.setInitialPageLoadName(pageName)
-
-  function generateError() {
-    throw new Error('Test Error in ' + pageName)
-  }
-  generateError()
-</script>
+        You can add webfonts, meta tags, or analytics to this file.
+        The build step will place the bundled scripts into the <body> tag.
 
+        To begin the development, run `npm start`.
+        To create a production bundle, use `npm run build`.
+      -->
+  </body>
+  <script>
+    function init(){
+      var pageName = window.location.pathname.split('/')[1]
+      window.elasticApm.setInitialPageLoadName(pageName)
```

Comment:
Not sure this would work, `setInitialPageLoadName` should be called before `load` event is triggered

Reply:
If you call it before the load, `window.elasticApm` is undefined and fail, it is why I have changed it to the download. With this change, there is no error in the console.

Reply:
Have you tried without setting this? We have a new optimized slug pattern matching that should get a better transaction name, we wouldn't need this anymore. 

Also as @kyungeunni mentioned, this needs to be called before `load`. So better would be to either use the `observe` API and call this on `transaction:end` or call this on `script onload` method.

Reply:
>So better would be to either use the observe API and call this on transaction:end

I am not sure where put this and how, so I will put a `<script onload="" >` call

Reply:
@kuisathaverat

You can also set the `setInitialPageLoadName` in the file where you initialize the RUM agent instead if you want

```
var apm = initApm(rumConfig)
apm.setInitialPageLoadName(window.location.pathname.split('/')[1])
```

The file is this one: https://github.com/elastic/opbeans-frontend/blob/main/src/rum.js

Reply:
Apologies for the delay, Saw that you have figured out the `onload` script.